nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/JerseyContainerCommandService.java

```diff
@@ -138,8 +140,15 @@ private JerseyContainer getJerseyContainer(ResourceConfig config) {
             ServiceFinder.setIteratorProvider(iteratorProvider);
             RestLogging.restLogger.log(Level.FINEST,
                 () -> this + ": Creating Jersey container for " + HttpHandler.class + " and " + config);
-            final HttpHandler httpHandler = ContainerFactory.createContainer(HttpHandler.class, config);
-            return new JerseyContainer() {
+            final GrizzlyHttpContainer httpHandler = ContainerFactory.createContainer(GrizzlyHttpContainer.class, config);
+            final ServiceLocator jerseyLocator = httpHandler.getApplicationHandler().getInjectionManager().getInstance(ServiceLocator.class);
+        /*
+            We enable a temporary distribution service until the HK2 Extras package is fixed so that we can enable
```

Comment:
What do we exactt need to fix in HK2 Extras?

Maybe good idea to create an issue here: https://github.com/eclipse-ee4j/glassfish-hk2-extra/issues ?

Reply:
Interesting btw that HK2 has the hierarchy of bean archives just as CDI has.

Reply:
> What do we exactt need to fix in HK2 Extras?

@arjantijms, PR already in description, eclipse-ee4j/glassfish-hk2#1058 ))

Reply:
Thanks! Was looking at HK2 Extras itself, but it was already there ;)